gen/PathPlannerTrajectory.yml

```diff
@@ -5,31 +5,69 @@ classes:
     methods:
       PathPlannerTrajectory:
         overloads:
-          std::vector<Waypoint>, units::meters_per_second_t, units::meters_per_second_squared_t, bool:
-          std::vector<PathPlannerState>:
+          std::vector<Waypoint>&, std::vector<EventMarker>&, PathConstraints, bool:
           "":
+      getStartStopEvent:
+      getEndStopEvent:
       sample:
       getStates:
+        overloads:
+          '[const]':
+          "":
+      getMarkers:
+        overloads:
+          '[const]':
+          "":
```

Comment:
There's a bunch of const-qualified overloads on these methods now. The const overloads return copies, whereas the non-const overloads return references. We need to decide which is the most appropriate.

Reply:
If it's returning a vector, it's always getting copied regardless whether it's a reference or not.

Reply:
Good point. Still need to figure out whether we should take a copy or reference for the `get*State` methods though.